cache/size_utils.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package cache
+
+import (
+	"math"
+	"reflect"
+)
+
+/*
+From src/runtime/hashmap.go:
```

Comment:
Also there's a potential `loadFactor` constant, but then you use the magic number below rather than a `loadFactor` constant. I'd rather you used the constant.